FitTheWall/Assets/ThreeDPoseUnityBarracuda/Scripts/VNectBarracudaRunner.cs

```diff
@@ -0,0 +1,374 @@
+    private int HeatMapCol_Cube;
+    private float ImageScale;
+
+    /// <summary>
+    /// Buffer memory has 2D heat map
+    /// </summary>
+    private float[] heatMap2D;
+
+    /// <summary>
+    /// Buffer memory has offset 2D
+    /// </summary>
+    private float[] offset2D;
+    
+    /// <summary>
+    /// Buffer memory has 3D heat map
+    /// </summary>
+    private float[] heatMap3D;
+    
+    /// <summary>
+    /// Buffer memory hash 3D offset
+    /// </summary>
+    private float[] offset3D;
+    private float unit;
+    
+    /// <summary>
+    /// Number of joints in 2D image
+    /// </summary>
+    private int JointNum_Squared = JointNum * 2;
+    
+    /// <summary>
+    /// Number of joints in 3D model
+    /// </summary>
+    private int JointNum_Cube = JointNum * 3;
+
+    /// <summary>
+    /// HeatMapCol * JointNum
+    /// </summary>
+    private int HeatMapCol_JointNum;
+
+    /// <summary>
+    /// HeatMapCol * JointNum_Squared
+    /// </summary>
+    private int CubeOffsetLinear;
+
+    /// <summary>
+    /// HeatMapCol * JointNum_Cube
+    /// </summary>
+    private int CubeOffsetSquared;
+
+    /// <summary>
+    /// For Kalman filter parameter Q
+    /// </summary>
+    public float KalmanParamQ;
+
+    /// <summary>
+    /// For Kalman filter parameter R
+    /// </summary>
+    public float KalmanParamR;
+
+    /// <summary>
+    /// Lock to update VNectModel
+    /// </summary>
+    private bool Lock = true;
+
+    /// <summary>
+    /// Use low pass filter flag
+    /// </summary>
+    public bool UseLowPassFilter;
+
+    /// <summary>
+    /// For low pass filter
+    /// </summary>
+    public float LowPassParam;
+
+    public Text Msg;
+    public float WaitTimeModelLoad = 10f;
+    private float Countdown = 0;
+    public Texture2D InitImg;
+
+    private void Start()
+    {
+        // Initialize 
+        HeatMapCol_Squared = HeatMapCol * HeatMapCol;
+        HeatMapCol_Cube = HeatMapCol * HeatMapCol * HeatMapCol;
+        HeatMapCol_JointNum = HeatMapCol * JointNum;
+        CubeOffsetLinear = HeatMapCol * JointNum_Cube;
+        CubeOffsetSquared = HeatMapCol_Squared * JointNum_Cube;
+
+        heatMap2D = new float[JointNum * HeatMapCol_Squared];
+        offset2D = new float[JointNum * HeatMapCol_Squared * 2];
+        heatMap3D = new float[JointNum * HeatMapCol_Cube];
+        offset3D = new float[JointNum * HeatMapCol_Cube * 3];
+        unit = 1f / (float)HeatMapCol;
+        InputImageSizeF = InputImageSize;
+        InputImageSizeHalf = InputImageSizeF / 2f;
+        ImageScale = InputImageSize / (float)HeatMapCol;// 224f / (float)InputImageSize;
+
+        // Disabel sleep
+        Screen.sleepTimeout = SleepTimeout.NeverSleep;
+
+        // Init model
+        _model = ModelLoader.Load(NNModel, Verbose);
+        _worker = WorkerFactory.CreateWorker(WorkerType, _model, Verbose);
+
+        StartCoroutine("WaitLoad");
+
+    }
+
+    private void Update()
+    {
+        if (!Lock)
```

Comment:
Normally we don't really write any logic in Update loops, we use functions. But I don't mind it to much here, since it's just 1 if statement